add  `hookActionValidateOrder` concrete use case example (#1543)
PrestaShop/docs: modules/concepts/hooks/list-of-hooks.md

            }

            public function hookActionValidateOrder($params)
            {
                $orderObject = $params['order'];
                $customerObject = $params['customer'];
                $hasValidParams = Validate::isLoadedObject($orderObject) && Validate::isLoadedObject($orderObject);

                if ($hasValidParams && !$this->customerAlreadyRewarded((int) $customerObject->id)) {

                   $hasConfiguredState = in_array((int) $orderObject->getCurrentState(), $this->getConfuredOrdersStatesIds());
                   $hasCustomerRequiredNbrOfTheOrderToReward = $this->getCustomerValidOrdersNbr((int) $customerObject->id) == $this->getRequiredNbrOfTheOrderToReward();

                   if($hasConfiguredState && $hasCustomerRequiredNbrOfTheOrderToReward) {
                       $customerReward = $this->createCustomerReward($customerObject, $orderObject);

                       if(Validate::isLoadedObject($customerReward)) {
                             $this->setAlreadyRewarded($customerObject);
                             $this->notifyCustomer($customerObject, $customerReward);
			     

> 			     //TODO : of course don't forget to log if something fail here :)

Reviewer: ```suggestion
			     //TODO : of course don't forget to log if something fails here :)
```

                       }
                   }
                }
            }

            protected function customerAlreadyRewarded(int $idCustomer): bool
            {
                //TODO : check if customer already rewarded
            }

            protected setAlreadyRewarded(): void
            {
                //TODO: set customer was rewarded
            }

            protected function getConfuredOrdersStatesIds(): array
            {
                //TODO : return array with configured states ids in your module
            }

            protedect function getCustomerValidOrdersNbr(int $idCustomer): int
            {
               //TODO : return number of total order valid by customer
            }

            protected function getRequiredNbrOfTheOrderToReward(): int
            {

>                //TODO : return configured requored number of the order to reward the customer

Reviewer: ```suggestion
               //TODO : return configured number of orders required to reward the customer
```

            }

            protected function createCustomerReward(Customer $customer, Order $order): ?CartRule
            {

>                //TODO: generate customer cart rule (according the order amount for example)

Reviewer: ```suggestion
               //TODO: generate customer cart rule (according to the order amount for example)
```
